java/java.editor/src/org/netbeans/modules/java/editor/resources/DefaultAbbrevs.xml

```diff
@@ -489,6 +493,25 @@ ${cursor}]]></code></codetemplate>
         }
      }
      ]]></code></codetemplate>
+
+    <!-- Record generation -->
+
+    <codetemplate abbreviation="prec"><code><![CDATA[${no-format}private record ${Type}(${cursor}) {}]]></code></codetemplate>
+    <codetemplate abbreviation="preci"><code><![CDATA[${no-format}private record ${Type}(${cursor}) implements ${classVar editable="false" currClassName default="getClass()"} {}]]></code></codetemplate>
+    <codetemplate abbreviation="rec"><code><![CDATA[${no-format}public record ${Type}(${cursor}) {}]]></code></codetemplate>
+    <codetemplate abbreviation="reci"><code><![CDATA[${no-format}public record ${Type}(${cursor}) implements ${classVar editable="false" currClassName default="getClass()"} {}]]></code></codetemplate>
```

Comment:
how about letting `rec` generate `record ${Type}(${cursor}) {}` and `purec` `public record ${Type}(${cursor}) {}`?

Since i found declaring records inside methods is quite common, those would have no `public`/`private` modifier.